crates/starknet/src/state/state_diff.rs

```diff
@@ -118,6 +121,59 @@ impl StateDiff {
     }
 }
 
+impl From<StateDiff> for ThinStateDiff {
+    fn from(value: StateDiff) -> Self {
+        let declared_classes: Vec<(Felt, Felt)> =
+            value.class_hash_to_compiled_class_hash.into_iter().collect();
+
+        // cairo 0 declarations
+        let cairo_0_declared_classes: Vec<Felt> = value.cairo_0_declared_contracts;
+
+        // storage updates (contract address -> [(storage_entry, value)])
+        let storage_updates: Vec<(ContractAddress, Vec<(PatriciaKey, Felt)>)> = value
+            .storage_updates
+            .into_iter()
+            .map(|(address, entries)| (address, entries.into_iter().collect()))
+            .collect();
+
+        // contract nonces
+        let nonces: Vec<(ContractAddress, Felt)> = value.address_to_nonce.into_iter().collect();
+
+        // deployed contracts (address -> class hash)
+        let deployed_contracts: Vec<(ContractAddress, Felt)> =
+            value.address_to_class_hash.into_iter().collect();
+
+        ThinStateDiff {
+            deployed_contracts: deployed_contracts
+                .into_iter()
+                .map(|(address, class_hash)| DeployedContract { address, class_hash })
+                .collect(),
+            declared_classes: declared_classes
+                .into_iter()
+                .map(|(class_hash, compiled_class_hash)| ClassHashes {
+                    class_hash,
+                    compiled_class_hash,
+                })
+                .collect(),
+            deprecated_declared_classes: cairo_0_declared_classes,
+            nonces: nonces
+                .into_iter()
+                .map(|(address, nonce)| ContractNonce { contract_address: address, nonce })
+                .collect(),
+            storage_diffs: storage_updates
+                .into_iter()
+                .map(|(contract_address, updates)| StorageDiff {
+                    address: contract_address,
+                    storage_entries: updates
+                        .into_iter()
+                        .map(|(key, value)| StorageEntry { key, value })
+                        .collect(),
+                })
+                .collect(),
+            replaced_classes: vec![],
```

Comment:
Should it be empty?

Reply:
I dont know to be honest, it was from old PR. But looking at `CommitmentStateDiff` there is no clear information about the replaced_classes

Reply:
Delegated to https://github.com/0xSpaceShard/starknet-devnet-rs/issues/252
Resolving